test/Microsoft.SqlTools.ServiceLayer.Test.Common/TestConnectionProfileService.cs

```diff
@@ -0,0 +1,206 @@
+                connectParams.Connection.TrustServerCertificate = false;
+            }
+            return connectParams;
+        }
+
+        private static IEnumerable<TestServerIdentity> InitTestServerNames()
+        {
+            try
+            {
+                string testServerNamesFilePath = Environment.GetEnvironmentVariable("TestServerNamesFile");
+                if (!string.IsNullOrEmpty(testServerNamesFilePath))
+                {
+                    string jsonFileContent = File.ReadAllText(testServerNamesFilePath);
+                    return Newtonsoft.Json.JsonConvert.DeserializeObject<IList<TestServerIdentity>>(jsonFileContent);
+                }
+                else
+                {
+                    return Enumerable.Empty<TestServerIdentity>();
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Failed to load the database connection server name settings. error: " + ex.Message);
+                return new List<TestServerIdentity>();
+            }
+        }
+
+        private static ConnectionSetting InitSetting()
+        {
+            try
+            {
+                string settingsFileContents = GetSettingFileContent();
+                ConnectionSetting setting = Newtonsoft.Json.JsonConvert.DeserializeObject<ConnectionSetting>(settingsFileContents);
+
+                return setting;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Failed to load the connection settings. error: " + ex.Message);
+                return new ConnectionSetting();
+            }
+        }
+
+        /// <summary>
+        /// Get the location of setting.json. Returns the value of environment variable 'SettingsFileName' and if it's empty returns
+        /// the location of vs code settings.json
+        /// </summary>
+        /// <returns></returns>
+        private static string GetSettingFileContent()
+        {
+            string settingsFilename;
+            settingsFilename = Environment.GetEnvironmentVariable("SettingsFileName");
+
+            if (string.IsNullOrEmpty(settingsFilename))
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    settingsFilename = Environment.GetEnvironmentVariable("APPDATA") + @"\Code\User\settings.json";
+                }
+                else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    settingsFilename = Environment.GetEnvironmentVariable("HOME") + @"/Library/Application Support/Code/User/settings.json";
+                }
+                else
+                {
+                    settingsFilename = Environment.GetEnvironmentVariable("HOME") + @"/.config/Code/User/settings.json";
+                }
+            }
+            string settingsFileContents = File.ReadAllText(settingsFilename);
```

Comment:
This could be null if Code isn't installed or the Insiders build is used. Please update to reflect this - e.g. error handling and indicate where to specify the file